### PR TITLE
chore: refactor ops package

### PR DESCRIPTION
**Pull request by mlavacca**
**What this PR does / why we need it**:

the ops package has been refactored with the following changes:
- all the mock files have been moved under `controller/konnect/ops/mock`
- the generated mock files have been prefixed with `zz_generated.`
- all the SKD interface files have been moved under `controller/konnect/ops/sdk`

**Which issue this PR fixes**

Fixes #

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

**Review by czeslavo — COMMENTED** [review]
Thanks for doing this. 👍  

One thing that I would maybe change is the location of mocks. As they're all SDK mocks, IMO it makes `ops/sdk/mock` package more suitable for them.

File: .mockery.yaml [hunk 1]
```diff
@@ -3,8 +3,8 @@ inpackage: True
 disable-version-string: True
 with-expecter: True
 
-filename: "{{ trimSuffix .InterfaceFile \".go\" | base | lower }}_mock.go"
-dir: "{{ .InterfaceDir }}"
+filename: "zz_generated.{{ trimSuffix .InterfaceFile \".go\" | base | lower }}_mock.go"
```
Comment (pmalek) [inline .mockery.yaml:5]: nit: most of our generated files have `zz_generated_` prefix. Any specific reason to use the dot here instead?

File: controller/konnect/ops/mock/sdkfactory_mock.go [hunk 2]
```diff
@@ -1,9 +1,11 @@
-package ops
+package mock
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
+
+	sdkops "github.com/kong/gateway-operator/controller/konnect/ops/sdk"
```
Comment (pmalek) [inline controller/konnect/ops/mock/sdkfactory_mock.go]: Might as well as this to `golangci-lint` config to enforce the same import alias: https://github.com/Kong/gateway-operator/blob/a44e9a9f0b4b4e03baca0347fb7ec66e6036d073/.golangci.yaml#L43-L67

**Review by pmalek — COMMENTED** [review]
Good effort! Thanks for working on this 🙇

**Comment by mlavacca** [comment]
> Thanks for doing this. 👍
> 
> One thing that I would maybe change is the location of mocks. As they're all SDK mocks, IMO it makes `ops/sdk/mock` package more suitable for them.

Good point, see 63f1a70c1ae18607f3863ab6a9cf1b4dafa53f48

File: .mockery.yaml (hunk 1 above)
Comment (mlavacca) [reply .mockery.yaml:5]: I've taken the dot format from https://github.com/kubernetes/kubernetes/blob/e39571591d2917ca642f47ebca11253a91793c4b/.generated_files#L17.

But I agree we should stick to one naming convention. I'd lean toward renaming alle our generated ones to stick to the upstream format, but can be done in a subsequent PR. WDYT?

Comment (pmalek) [reply .mockery.yaml:5]: Agreed. 👍

File: controller/konnect/ops/mock/sdkfactory_mock.go (hunk 2 above)
Comment (mlavacca) [reply controller/konnect/ops/mock/sdkfactory_mock.go]: https://github.com/Kong/gateway-operator/pull/764/commits/028ab7a5ab2e87a0d5dd1187cb2ba42b6f0b0854

File: .mockery.yaml (hunk 1 above)
Comment (mlavacca) [reply .mockery.yaml:5]: https://github.com/Kong/gateway-operator/issues/766

**Comment by pmalek** [comment]
Linking #766 as related.

**Review by pmalek — APPROVED** [review]

**Pull request merged**